.github/workflows/e2e.yml

```diff
@@ -0,0 +1,131 @@
+name: E2E Test
+
+on:
+  pull_request:
+    branches:
+      - main
+      - "release-*"
+    paths-ignore:
+      - "docs/**"
+      - "**.md"
+      - "sec-scanners-config.yaml"
+
+env:
+  NATS_MANAGER_IMAGE: europe-docker.pkg.dev/kyma-project/prod/nats-manager
+  EVENTING_MANAGER_IMAGE: europe-docker.pkg.dev/kyma-project/prod/eventing-manager
+  EPP_IMAGE: europe-docker.pkg.dev/kyma-project/dev/eventing-publisher-proxy:PR-${{ github.event.number }}
+
+jobs:
+  e2e-nats:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout EPP repository
+        uses: actions/checkout@v4
+        with:
+          path: main
+
+      - name: Checkout EM repository
+        uses: actions/checkout@v4
+        with:
+          repository: kyma-project/eventing-manager
+          path: eventing-manager
+
+      - name: Checkout NM repository
+        uses: actions/checkout@v4
+        with:
+          repository: kyma-project/nats-manager
+          path: nats-manager
+
+      - name: Cache binaries
+        id: cache-binaries
+        uses: actions/cache@v3
+        with:
+          path: bin
+          key: ${{ runner.os }}-bin
+
+      - name: Install k3d tools
+        run: |
+          make -C eventing-manager/hack/ci/ install-k3d-tools
+
+      - name: Install Kyma CLI & setup k3d cluster using Kyma CLI
+        run: |
+          make -C eventing-manager kyma
+          make -C main/hack/ci/ create-k3d
+          kubectl version
+          kubectl cluster-info
+
+      - name: Create Namespace
+        run: |
+          kubectl create ns kyma-system
+
+      - name: Setup NATS Manager
+        run: |
+          make -C nats-manager install
+
+      - name: Deploy NATS Manager
+        run: |
+          version=$(curl -i https://github.com/kyma-project/nats-manager/releases/latest | awk -F '/' '/^location/ {print  substr($NF, 1, length($NF)-1)}')
+          make -C nats-manager deploy IMG=$NATS_MANAGER_IMAGE:v20231013-6091000e
```

Comment:
`v20231013-6091000e` is hardcoded until we have a new tagged release of NATS Manager